### PR TITLE
add dispatch and useSelector to use Redux in Components

### PR DESCRIPTION
**Pull request by tiagomarin**
In this pull request I added the necessary methods of React Redux to use the Redux store in the components.

Also I added the funcionality for removing and adding books to the list by dispatching actions and using the state connected to Redux store. All books have a unique ID, for that I used the npm package [uuid](https://www.npmjs.com/package/uuid).

Lastly I added the funcionality for the button "status" on the categories page to display a message "under construction" below the button.

### Please note
I created the reducers and actions using 2 diferent approaches, **one with normal Redux** and the other using the **createSlice from the toolkit library**. I plan to make both work to practice both, but at the end I'll choose 1 of them to leave in the project.

### CODE REVIEW

File: src/Pages/BooksPage.js [hunk 1]
```diff
@@ -1,24 +1,29 @@
-import './BooksPage.css';
-import bookList from '../fakeBooksData';
+import { useSelector } from 'react-redux';
 import Book from '../components/Book';
-import AddBook from '../components/AddBook';
+import AddBookForm from '../components/AddBookForm';
+// import { removeBook } from '../Redux/book/book';
```
Comment (Nemwel-Boniface) [inline src/Pages/BooksPage.js:4]: - [ ] [ OPTIONAL]  Kindly remove all commented-out and unwanted/ unused pieces of code from your code base. It makes it look messy and cluttered.

File: src/Pages/BooksPage.js [hunk 2]
```diff
@@ -1,24 +1,29 @@
...
+    <>
+      <ul className="books-list">
+        {bookList.map((book) => (
+          <Book
+            key={book.id}
+            id={book.id}
+            author={book.author}
+            title={book.title}
+            category={book.category}
+            // removeClickHandler={removeClickHandler}
```
Comment (Nemwel-Boniface) [inline src/Pages/BooksPage.js:20]: - [ ] [ OPTIONAL]  Kindly remove all commented-out and unwanted/ unused pieces of code from your code base. It makes it look messy and cluttered.

**Review by Nemwel-Boniface — APPROVED** [review]
## Hello @tiagomarin              :wave: ,
I am Nemwel your reviewer for this PR :man_beard:  

Thank you for meeting all the project requirements good job! :partying_face: 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! :partying_face: 🎉 



## Highlights

Great job setting up the correct linters ✔️ 
Professional documentation ✔️ 
Correct workflow ✔️ 
Meaningful commit messages ✔️ 
Met all the milestone requirements ✔️ 

## [Optional] suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@Nemwel-Boniface** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Pages/BooksPage.js (hunk 1 above)
Comment (tiagomarin) [reply src/Pages/BooksPage.js:4]: Thanks for pointing that, I forgot to remove!

File: src/Pages/BooksPage.js (hunk 2 above)
Comment (tiagomarin) [reply src/Pages/BooksPage.js:20]: Thanks for pointing that, I forgot to remove!

**Pull request merged**